File: Backend/intoTheAbyss/src/network/packets/PlayerLocationPacket.java
package network.packets;

public class PlayerLocationPacket {
	private String playerID;
	private int playerLocationFloor, playerPositionX, playerPositionY;

	public PlayerLocationPacket() {

	}

	public String getPlayerID() {
		return playerID;
	}

	public int getPlayerFloor() {
		return playerLocationFloor;
	}

	public int getXPos() {
		return playerPositionX;
	}

	public int getYPos() {
		return playerPositionY;
	}

}

File: Frontend/IntotheAbyss/app/src/main/java/com/example/intotheabyss/networking/packets/PlayerLocationPacket.kt
package com.example.intotheabyss.networking.packets

/**
 * The packet that gets sent to the server whenever the user's player moves. The client then sends the information to
 * the packet to notify other concurrent players and update monster logic.
 *
 * @constructor Given the correct parameters, sets the parameters as the payload for this packet.
 * @param playerID The ID of the player moving.
 * @param playerLocationFloor The current floor the player is on.
 * @param playerPositionX The x-position of the player.
 * @param playerPositionY The y-position of the player.
 *
 * @author Benjamin Vogel
 */
class PlayerLocationPacket(var playerID: String, var playerLocationFloor: Int,
                           var playerPositionX: Int, var playerPositionY: Int)